service/siteService.js

```diff
@@ -78,3 +89,35 @@ transformHours = function(time) {
 	}
 }
 
+transformPublicTransit = function(publicTransitOptions) {
+	return publicTransitOptions ? getPublicTransit(publicTransitOptions) : undefined
+}
+
+getPublicTransit = function(publicTransitOptions) {
+	let options = []
+	if(publicTransitOptions) {
+		publicTransitOptions.forEach(function(transitOption) {
+			const properties = transitOption.split("-")
+			const route = getTransitOption(properties)
+			options.push(route)
+		})
+	}
+	return options
+}
+
+getTransitOption = function(properties) {
+	const routeName = properties[0]
+	const type = properties[1]
+	const distance = properties[2]
+
+	switch(type) {
```

Comment:
I wanted to do an enum for this but there didn't seem to be a good way of doing that in javascript. I played around with it a bit but in the end a switch seemed to be the cleanest and most straightforward. Thoughts?

Reply:
Last I looked at this in either JS or Python, the sanest way to maintain for me would be by defining a map as a constant for the wonky bits:

```
TRANSIT_VISUALS = {
  BLUELINE: {backgroundColor: ...},
  GREENLINE: {backgroundColor: ...},
  ...
}
```

... then below you can just call

`return Object.assign({}, {routeName: ...}, TRANSIT_VISUALS[type]);`

That said, I wouldn't worry about it unless the transit options start growing by quite a bit.

Reply:
Yeah, I actually tried that but was having issues with the `TRANSIT_VISUALS[type]` piece, I'm sure I was just doing something a bit wrong - but yeah, I agree not worth worrying about too much. I don't expect the transit options to grow much outside of this at this point.. if anything maybe adding another light rail if that becomes a thing...should know more later this week as we start actually getting the data for it though